File: src/main/java/com/workforce/hrm/service/NotificationService.java
package com.workforce.hrm.service;

import java.util.List;

import com.workforce.hrm.dto.request.NotificationRequestDTO;
import com.workforce.hrm.dto.response.NotificationResponseDTO;

public interface NotificationService {

	// Create a notification
	NotificationResponseDTO createNotification(NotificationRequestDTO requestDTO);

	// Get all notifications of a user
	List<NotificationResponseDTO> getNotificationsByUser(Long userId);

	// Get unread notifications of a user
	List<NotificationResponseDTO> getUnreadNotifications(Long userId);

	// Get unread notification count
	long getUnreadNotificationCount(Long userId);

	// Mark notification as READ
	NotificationResponseDTO markAsRead(Long notificationId);

	// Delete notification
	void deleteNotification(Long notificationId, Long userId);

}

File: hrm-portal-frontend/src/services/notificationService.js
import api from "./api";

const notificationService = {

  async getNotifications({
    page = 0,
    size = 20,
    unreadOnly = false
  } = {}) {

    const response = await api.get("/notifications");
    const notifications = response.data || [];

    return unreadOnly
      ? notifications.filter(({ status }) => status === "UNREAD")
      : notifications.slice(page * size, (page + 1) * size);
  },


  async getUnreadCount() {

    const response = await api.get(
      "/notifications/unread-count"
    );

    return response.data;
  },


  async markAsRead(notificationId) {

    if (!notificationId) {
      throw new Error(
        "Notification ID is required."
      );
    }

    const response = await api.put(
      `/notifications/${notificationId}/read`
    );

    return response.data;
  },


  async markAllAsRead() {

    const response = await api.put(
      "/notifications/read-all"
    );

    return response.data;
  },


  async deleteNotification(
    notificationId
  ) {

    if (!notificationId) {
      throw new Error(
        "Notification ID is required."
      );
    }

    const response = await api.delete(`/notifications/${notificationId}`);

    return response.data;
  }

};

export default notificationService;
